src/cosl/coordinated_workers/coordinator.py

```diff
@@ -236,6 +237,7 @@ def __init__(
             workload_tracing_protocols: A list of protocols that the worker intends to send
                 workload traces with.
             catalogue_item: A catalogue application entry to be sent to catalogue.
+            worker_config_version: A function returning the workload version for which the generated worker config is intended.
```

Comment:
It feels strange to have this in the constructor.
And why is it a callable?
when the coordinator initializes, we might not yet know what worker config version we need to generate: the worker may still need to tell that to us.

So I'd rather expect the coordinator to expose a `get_requested_worker_version()` method for the caller (the charm) to know what worker version we're connected to

Reply:
besides, unless there's precisely one workload version that the workers are requesting, we'll block and do nothing, so there's no point in taking this as an argument from the charm. 
We'll send back the same version we take in, unless we can't satisfy it.

Reply:
so what I think we need instead is:

`self._workers_config_getter = lambda: workers_config(self, version=self.get_requested_worker_version())`

so the charm can implement:

`def generate_worker_config(self, coordinator, version):`

Reply:
may need something smart for that to be backwards compatible, like checking the number of args that the config generator takes